crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -108,6 +108,31 @@ impl<T: TransactionOrdering> TxPool<T> {
         self.all().txs_iter(sender).last().map(|(_, tx)| Arc::clone(&tx.transaction))
     }
 
+    /// Returns the next valid transaction for the given sender.
+    pub(crate) fn get_next_valid_transaction_by_sender(
+        &self,
+        sender: SenderId,
+        on_chain_nonce: u64,
+    ) -> Option<Arc<ValidPoolTransaction<T::Transaction>>> {
+        let mut current_nonce = on_chain_nonce;
+        let mut last_connected_tx = None;
+
+        for (_, tx) in self.all().txs_iter(sender) {
+            let tx_nonce = tx.transaction.nonce();
+            
+            if tx_nonce == current_nonce + 1 {
+                // This transaction is connected to the previous one
+                last_connected_tx = Some(Arc::clone(&tx.transaction));
+                current_nonce = tx_nonce;
+            } else if tx_nonce > current_nonce + 1 {
+                // We've found a gap, so we stop here
+                break;
+            }
```

Comment:
we can make this slightly nicer by incrementing the next consecutive nonce early